src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/ThingRegistry.java

```diff
@@ -116,32 +116,27 @@ private Thing storeThing(Thing thing) {
     /**
      * Returns whether the Thing is associated to the given IoT Certificate.
      * Returns valid locally registered result when IoT Core cannot be reached.
-     * TODO: add a separate refreshable caching layer for offline auth
      *
      * @param thing IoT Thing
      * @param certificate IoT Certificate
      * @return whether thing is attached to the certificate
      * @throws CloudServiceInteractionException when thing <-> certificate association cannot be verified
      */
     public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate) {
-        try {
-            if (iotAuthClient.isThingAttachedToCertificate(thing, certificate)) {
-                thing.attachCertificate(certificate.getCertificateId());
-                updateThing(thing);
-                return true;
-            } else {
-                thing.detachCertificate(certificate.getCertificateId());
-                updateThing(thing);
-            }
-        } catch (CloudServiceInteractionException e) {
-            // TODO: also check the instant when this cert was attached this thing
-            //  and validate against configured offline TTL
-            if (thing.isCertificateAttached(certificate.getCertificateId())) {
-                return true;
-            }
-            throw e;
+        // If we have a local cache hit, then return true. Else, go to cloud
+        if (thing.isCertificateAttached(certificate.getCertificateId())) {
```

Comment:
This doesn't feel like an operation that should be part of the registry. This has some business logic embedded into it. a Domain Service seems like a better fit to it (if we don't want to do this now, maybe add a TODO, so we can take car of it later).

The update portion of this is part of the registry that is good but this if conditionals that decide to attach certificates to a thing are not. This is breaking CQS

Reply:
Yes, completely agree. Our session factory is already calling this though, so I kept it as is, for now. Can refactor out later